src/auth/auth-config.ts

```diff
@@ -1451,3 +1451,141 @@ export class OIDCConfig implements OIDCAuthProviderConfig {
     };
   }
 }
+
+/**
+ * The request interface for updating a SMS Region Config.
+ * Configures the regions where users are allowed to send verification SMS.
+ * This is based on the calling code of the destination phone number.
+ */
+export type SmsRegionConfig = AllowByDefaultWrap | AllowlistOnlyWrap;
+
+/**
+ * Mutual exclusive SMS Region Config of AllowByDefault interface
+ */
+export interface AllowByDefaultWrap {
+  /**
+   * Allowing every region by default.
+   */
+  allowByDefault: AllowByDefault;
+  /** @alpha */
+  allowlistOnly?: never;
+}
+
+/**
+ * Mutual exclusive SMS Region Config of AllowlistOnly interface
+ */
+export interface AllowlistOnlyWrap {
+  /**
+   * Only allowing regions by explicitly adding them to an
+   * allowlist.
+   */
+  allowlistOnly: AllowlistOnly;
+  /** @alpha */
+  allowByDefault?: never;
+}
+
+/**
+ * Defines a policy of allowing every region by default and adding disallowed
+ * regions to a disallow list.
+ */
+export interface AllowByDefault {
+  /**
+   * Two letter unicode region codes to disallow as defined by
+   * https://cldr.unicode.org/
+   * The full list of these region codes is here:
+   * https://github.com/unicode-cldr/cldr-localenames-full/blob/master/main/en/territories.json
+   */
+  disallowedRegions: string[];
+}
+
+/**
+ * Defines a policy of only allowing regions by explicitly adding them to an
+ * allowlist.
+ */
+export interface AllowlistOnly {
+    /**
+   * Two letter unicode region codes to allow as defined by
+   * https://cldr.unicode.org/
+   * The full list of these region codes is here:
+   * https://github.com/unicode-cldr/cldr-localenames-full/blob/master/main/en/territories.json
+   */
+  allowedRegions: string[];
+}
+
+export class SmsRegionsAuthConfig {
```

Comment:
I believe this is an internal class. Let's add a description and mark it `@internal`.
Ex:
```
/**
 * Defines the OIDCConfig class used to convert a client side configuration to its
 * server side representation.
 *
 * @internal
 */
 ```

Reply:
Updated